sycl/doc/design/CompileTimeProperties.md

```diff
@@ -465,25 +465,48 @@ When the device compiler generates code to reference the decorated member
 variable, it emits a call to the LLVM intrinsic function
 [`@llvm.ptr.annotation`][10] that annotates the pointer to that member
 variables, similar to the way the existing `[[clang::annotate()]]` attribute
-works.  Illustrating this with some simplified LLVM IR that matches the example
-code above:
+works.
 
-[10]: <https://llvm.org/docs/LangRef.html#llvm-ptr-annotation-intrinsic>
+The front-end encodes the properties from the C++ attribute
+`[[__sycl_detail__::add_ir_annotations_member()]]` into the
+`@llvm.ptr.annotation` call as follows:
+
+* The first parameter to `@llvm.ptr.annotation` is the pointer to annotate (as
+  with any call to this intrinsic).
+* The second parameter is the literal string `"sycl-properties"`.
+* The third parameter is the name of the source file (as with any call to this
+  intrinsic).
+* The fourth parameter is the line number (as with any call to this intrinsic).
+* The fifth parameter is a pointer to a constant global variable. The type of
```

Comment:
Is it allowed adding parameters to the `@llvm.ptr.annotation` intrinsic? In the [LangRef](https://llvm.org/docs/LangRef.html#llvm-ptr-annotation-intrinsic) only 4 parameters are described.

Reply:
That is curious. According to the intrinsic definition in [Intrinsics.td](https://github.com/intel/llvm/blob/sycl/llvm/include/llvm/IR/Intrinsics.td#L1015) it has 5 parameters, the last being another `i8 *`. It does not seem to be a recent change so I think maybe someone forgot to update the LangRef?